### ISSUE TITLE
Support for CWT

### ISSUE THREAD

**Issue by pdxjohnny** [ucan-wg/spec#30]
It would be great if ucan could support CWT as well as JWT for embedded us cases

References:
- CBOR spec: [RFC 8949](https://www.rfc-editor.org/rfc/rfc8949.html)
- CWT spec: [RFC 8392](https://www.rfc-editor.org/rfc/rfc8392.html)

**Comment by expede** [comment]
Hey @pdxjohnny; thanks for the idea! There's some related discussion here: https://github.com/ucan-wg/spec/discussions/18

**Comment by expede** [comment]
The TL;DR on current thinking (and we *can* be convinced otherwise!) is that this spec is the one for interoperability between systems. If we introduce many other encodings, then we run into token incompatibilities, or larger implementations that support multiple codecs. If folks want to adapt the spec to an internal use case that uses a different encoding, that's a great idea and we're happy to help, but that we can't expect it to interop with other systems and so shouldn't be in the spec directly.

THAT SAID, please do comment in that discussion, and especially with any concrete tradeoffs that you'd like to solve for 😄

**Comment by pdxjohnny** [comment]
Thanks @expede! Just commented: https://github.com/ucan-wg/spec/discussions/18#discussioncomment-2283702

Also, credit where credit is due, this idea came from @nedmsmith :)

**Comment by expede** [comment]
Awesome; thanks both of you! Also, always super interested to hear about new use cases. If not something that can be public in text, I'm always happy to hop on a call sometime 😃

**Comment by expede** [comment]
Hey @pdxjohnny 👋 In principle, if you're willing to deterministically encode the CWT, then you should be able to use the [canonicalization spec](https://github.com/ucan-wg/canonicalization) and/or [ucan-ipld](https://github.com/ucan-wg/ucan-ipld) to convert to/from CWT.

Does that meet your CWT needs?
